modules/basics/src/main/java/com/opengamma/strata/basics/date/DateSequence.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.basics.date;
+
+import static com.opengamma.strata.basics.date.LocalDateUtils.plusDays;
+
+import java.time.LocalDate;
+
+import org.joda.convert.FromString;
+import org.joda.convert.ToString;
+
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.named.ExtendedEnum;
+import com.opengamma.strata.collect.named.Named;
+
+/**
+ * A series of dates identified by name.
+ * <p>
+ * This interface encapsulates a sequence of dates as used in standard financial instruments.
+ * The most common are the quarterly IMM dates, which are on the third Wednesday of the March,
+ * June, September and December.
+ * <p>
+ * The most common implementations are provided in {@link DateSequences}.
+ * <p>
+ * Note that the dates produced by the sequence may not be business days.
+ * The application of a holiday calendar is typically the callers responsibility.
+ * <p>
+ * All implementations of this interface must be immutable and thread-safe.
+ */
+public interface DateSequence
+    extends Named {
+
+  /**
+   * Obtains a {@code DateSequence} from a unique name.
+   * 
+   * @param uniqueName  the unique name of the sequence
+   * @return the date sequence
+   * @throws IllegalArgumentException if the name is not known
+   */
+  @FromString
+  public static DateSequence of(String uniqueName) {
+    ArgChecker.notNull(uniqueName, "uniqueName");
+    return extendedEnum().lookup(uniqueName);
+  }
+
+  /**
+   * Gets the extended enum helper.
+   * <p>
+   * This helper allows instances of {@code DateSequence} to be lookup up.
+   * It also provides the complete set of available instances.
+   * 
+   * @return the extended enum helper
+   */
+  public static ExtendedEnum<DateSequence> extendedEnum() {
+    return DateSequences.ENUM_LOOKUP;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Finds the next date in the sequence, always returning a later date.
+   * <p>
+   * Given an input date, this method returns the next date after it from the sequence.
+   * 
+   * @param date  the input date
+   * @return the next sequence date after the input date
+   * @throws IllegalArgumentException if there are no more sequence dates
+   */
+  public default LocalDate next(LocalDate date) {
+    ArgChecker.notNull(date, "date");
+    LocalDate next = plusDays(date, 1);
+    return nextOrSame(next);
+  }
+
+  /**
+   * Finds the next date in the sequence, returning the input date if it is a date in the sequence.
+   * <p>
+   * Given an input date, this method returns a date from the sequence.
+   * If the input date is in the sequence, it is returned.
+   * Otherwise, the next date in the sequence is returned.
```

Comment:
For clarity maybe it should say "the next date in the sequence after the input date..."